### PR TITLE
tests: buidler gas report

### PR DESCRIPTION
**Pull request by tynes**
## Description

Adds an autogenerated gas report at the end of `yarn test`.
This was super simple to add, literally just importing something into the buidler config.
Maybe we want something different? It would be nice to have a way to know changes in the amount of gas used for any change in the contracts. This won't give that exactly, but is a step in the right direction.

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: buidler.config.ts [hunk 1]
```diff
@@ -11,7 +13,7 @@ usePlugin('buidler-typechain')
 
 import './plugins/buidler/ovm-compiler'
 
-const config: BuidlerConfig = {
+const config = {
```
Comment (tynes) [inline buidler.config.ts:16]: Had to remove the type from this as typescript would yell

**Comment by tynes** [comment]
This does add quite a few dependencies, not sure if worth it. Thoughts?

File: buidler.config.ts [hunk 2]
```diff
@@ -29,6 +31,9 @@ const config: BuidlerConfig = {
     outDir: 'build/types',
     target: 'ethers-v5',
   },
+  gasReporter: {
+    currency: 'USD',
```
Comment (ben-chain) [inline buidler.config.ts:35]: pricing this in non-gas units seems like a very strange choice.

**Comment by ben-chain** [comment]
Can this tool give us easy ways to add explicit assertions in tests about gas?  I think this is what we really need; to catch regressions in CI/testing.

File: buidler.config.ts (hunk 2 above)
Comment (TransmissionsDev) [reply buidler.config.ts:35]: it shows both gas units and usd price :p

**Comment by gakonst** [comment]
As I wrote in chat, this should be prob closed in favor of gas snapshots?

**Pull request closed without merging**